.github/workflows/check_for_redirects.yml

```
@@ -0,0 +1,39 @@
+name: Redirects Workflow
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, labeled, unlabeled]
+jobs:
+  checkIfRedirectsAreNeeded:
+    permissions:
+      pull-requests: write # used to add label
+      contents: read
+    name: Check if redirects are needed
+    runs-on: ubuntu-latest
+    if: github.event.action == 'opened' || github.event.action == 'synchronize'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+      - name: Get count of deleted files
+        run: |
+          git fetch origin main
+          echo "DELETED_FILES=$(git diff --name-status --diff-filter=D origin/main -- src/pages src/fragments ${{ github.pull_request.head.sha }} -- src/pages src/fragments | wc -l)" >> $GITHUB_ENV
+      - name: Add redirects-needed label if there are deleted files
+        if: ${{ env.DELETED_FILES > 0 }}
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { issue: { number: issue_number }, repo: { owner, repo } } = context;
+            const labels = ['redirects-needed'];
+            github.rest.issues.addLabels({ owner, repo, issue_number, labels});
+  blockPrIfRedirectsNeeded:
+    name: Block PR if redirects are needed
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'redirects-needed')
+    steps:
+      - name: Exit with error
+        run: exit 1
```

Wonder if we can move this logic to the previous job:

```suggestion
      - name: Add redirects-needed label if there are deleted files
        if: ${{ env.DELETED_FILES > 0 }}
        uses: actions/github-script@v6
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          script: |
            const { issue: { number: issue_number }, repo: { owner, repo } } = context;
            const labels = ['redirects-needed'];
            github.rest.issues.addLabels({ owner, repo, issue_number, labels});
    - name: Block PR if redirects are needed
       if: ${{ env.DELETED_FILES > 0 }}
       run: exit 1
```

That way, we don't have to rely on `github.event.pull_request.labels.*.name` on updating dynamically after previous step runs. (Correct me if I'm wrong, but I imagine `github.event` object is static once the workflow starts).

---

Thank you so much for taking a look William! I think I need to keep this as separate action after my latest changes. I ended up splitting the `git diff` checks to have it run on when a PR is opened and each time a sync happens. This is so I can keep checking if a new push introduces a new deletion.

I am no longer adding the `redirects-label` from my workflow and instead am using Mergify to add the label when any of these jobs fail. 

This workflow also runs on the `labeled` event and so now I have a status check that will keep failing as long as the label is there. Right now the way to get it to pass is to have the team manually remove it so that the workflow runs again causing the job to get "skipped"

Hopefully all that makes sense, but if it doesn't please feel free to ask any questions you might have